core/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -256,7 +257,7 @@ public void addIndexStore(String type, Function<IndexSettings, IndexStore> provi
      * @param name Name of the SimilarityProvider
      * @param similarity SimilarityProvider to register
```

Comment:
I thought about this too but I figured that given it is temporary (i.e. this will be again a `BiFunction` after the cleanup task #22298 is done) it might be ok. But I might need to ponder once more.